types/src/transaction/transaction.rs

```diff
@@ -264,6 +268,77 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn apply_bitmask(
+    bitmask: u128,
+    mut outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Vec<AssetTransferOutput> {
+    match bitmask {
+        0 => return result,
+        _ => {
+            if ((bitmask & 0x1) == 1) && !outputs.is_empty() {
```

Comment:
I think it must be the error if the bit is set but the corresponding output does not exist.